### PR TITLE
New cosmetic robot types

### PR DESCRIPTION
**Pull request by Blundir**
## About The Pull Request

Adds 4 new cosmetic squad coloured robot types, sprites by The Lord#9786, with small edits by me.
dreamseeker_z8eatjjXHM
dreamseeker_YaSyVRk5kP
dreamseeker_sE6sFESpfT
dreamseeker_AX1uXB4dBD
dreamseeker_ybPH3UNS6u

You can pick up robot type at character editor
QDw898Vp5e

Also new robot types have corresponding sprites for huds
dreamseeker_hvRa58CZLh

## Why It's Good For The Game

More variety to a customization poor race.

## Changelog

:cl:
add: Added 4 new cosmetic robot types
/:cl:

### CODE REVIEW

**Comment by Coastfront** [comment]
While I do not like these sprites, even a mound of dirt stands taller than a voided flat plane. And they look better than what already exists.

god please allow robots to sculpt even a pitable face beyond their preset look

**Comment by zee11zee** [comment]
Alpha is looking excellent, I would like to say if this is going through I will be a Robo main

**Comment by Kuro020** [comment]
Gonna have to test these with the new robot armor sprites before i approve it

**Comment by Blundir** [comment]
> Gonna have to test these with the new robot armor sprites before i approve it

I can just make sprite sheets for it, in case there'd be any problem

**Comment by GreatKingWombat** [comment]
Any way to add an auto option, if you don't have a preferred squad set?

**Comment by Blundir** [comment]
> Any way to add an auto option, if you don't have a preferred squad set?

They aren't connected to squad code in any way and their squad aesthetics are trivial, nothing stops you from using non corresponding ones.

**Comment by Al-1ce** [comment]
> They aren't connected to squad code in any way and their squad aesthetics are trivial, nothing stops you from using non corresponding ones.


It shall remain an unwritten rule.

**Review by Kuro020 — APPROVED** [review]

**Comment by Ragecrabbo** [comment]
These look really nice although im not a fan of the red bandana thing (eye?) on the blue robot and the skins kinda blend in to the background floor sprites too much but otherwise these are solid quality and i really enjoy the animated visors.

File: code/datums/jobs/job/terragov.dm [hunk 1]
```diff
@@ -18,8 +18,21 @@
...
+					return /mob/living/carbon/human/species/robot/alpharii
+			if(prefs && prefs.robot_type == "Chilvaris")
+				if(GLOB.join_as_robot_allowed)
+					return /mob/living/carbon/human/species/robot/charlit
+			if(prefs && prefs.robot_type == "Ratcher")
+				if(GLOB.join_as_robot_allowed)
+					return /mob/living/carbon/human/species/robot/deltad
+			if(prefs && prefs.robot_type == "Sterling")
+				if(GLOB.join_as_robot_allowed)
+					return /mob/living/carbon/human/species/robot/bravada
```
Comment (TiviPlus) [inline code/datums/jobs/job/terragov.dm:35]: You can just early return here with (GLOB.join_as_robot_allowed)

File: code/game/objects/items/storage/backpack.dm [hunk 2]
```diff
@@ -7,7 +7,12 @@
 	desc = "You wear this on your back and put items into it."
 	icon_state = "backpack"
 	item_state = "backpack"
-	sprite_sheets = list("Combat Robot" = 'icons/mob/species/robot/backpack.dmi')
+	sprite_sheets = list(
+		"Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Sterling Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Chilvaris Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Hammerhead Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Ratcher Combat Robot" = 'icons/mob/species/robot/backpack.dmi')
```
Comment (TiviPlus) [inline code/game/objects/items/storage/backpack.dm:15]:
```suggestion
		"Hammerhead Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
		"Ratcher Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
		)
```

File: code/game/objects/items/storage/holsters.dm [hunk 3]
```diff
@@ -90,7 +90,12 @@
 /obj/item/storage/holster/backholster
 	name = "backpack holster"
 	desc = "You wear this on your back and put items into it. Usually one special item too."
-	sprite_sheets = list("Combat Robot" = 'icons/mob/species/robot/backpack.dmi')
+	sprite_sheets = list(
+		"Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Sterling Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Chilvaris Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Hammerhead Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Ratcher Combat Robot" = 'icons/mob/species/robot/backpack.dmi')
```
Comment (TiviPlus) [inline code/game/objects/items/storage/holsters.dm:98]:
```suggestion
		"Hammerhead Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
		"Ratcher Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
		)
```

File: code/game/objects/items/storage/holsters.dm [hunk 4]
```diff
@@ -130,7 +135,12 @@
 		/obj/item/ammo_magazine/rocket,
 		/obj/item/weapon/gun/launcher/rocket/recoillessrifle,
 	)
-	sprite_sheets = list("Combat Robot" = 'icons/mob/species/robot/backpack.dmi') //robots have their own snowflake back sprites
+	sprite_sheets = list(
+		"Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Sterling Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Chilvaris Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Hammerhead Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
+		"Ratcher Combat Robot" = 'icons/mob/species/robot/backpack.dmi') //robots have their own snowflake back sprites
```
Comment (TiviPlus) [inline code/game/objects/items/storage/holsters.dm:143]:
```suggestion
		"Ratcher Combat Robot" = 'icons/mob/species/robot/backpack.dmi',
		)
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Some parts are stinky but its not your fault fix these and its fine

**Pull request merged**